be/src/storage/runtime_filter_predicate.h

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include "column/chunk.h"
+#include "column/vectorized_fwd.h"
+#include "common/status.h"
+#include "exprs/runtime_filter_bank.h"
+
+namespace starrocks {
+class RuntimeFilterProbeDescriptor;
+
+class RuntimeFilterPredicate {
+public:
+    RuntimeFilterPredicate(RuntimeFilterProbeDescriptor* desc, ColumnId column_id)
+            : _rf_desc(desc), _column_id(column_id) {}
+    virtual ~RuntimeFilterPredicate() = default;
+
+    virtual Status evaluate(Chunk* chunk, uint8_t* selection, uint16_t from, uint16_t to);
+    virtual StatusOr<uint16_t> evaluate(Chunk* chunk, uint16_t* sel, uint16_t sel_size, uint16_t* target_sel);
+
+    ColumnId get_column_id() const { return _column_id; }
+    RuntimeFilterProbeDescriptor* get_rf_desc() const { return _rf_desc; }
+    bool init(int32_t driver_sequence);
+
+protected:
+    RuntimeFilterProbeDescriptor* _rf_desc;
+    const JoinRuntimeFilter* _rf = nullptr;
+    ColumnId _column_id;
```

Comment:
do we support runtime filter whose probe_expr is other Expr except ColumnRef  and/or multi-column partition exprs?

Reply:
not supported now, we can consider it in the future.